pkg/crypto/luks.go

```diff
@@ -24,9 +24,9 @@ func luksClose(volume string) (stdout string, err error) {
 	return cryptSetup("luksClose", volume)
 }
 
-func luksFormat(devicePath, passphrase string) (stdout string, err error) {
+func luksFormat(devicePath, passphrase, keyCipher, keyHash, keySize, pbkdf string) (stdout string, err error) {
 	return cryptSetupWithPassphrase(passphrase,
-		"-q", "luksFormat", "--type", "luks2", "--hash", "sha256",
+		"-q", "luksFormat", "--type", "luks2", "--cipher", keyCipher, "--hash", keyHash, "--key-size", keySize, "--pbkdf", pbkdf,
```

Comment:
Would this break the existing encrypted volumes w/o these options? Should we set these options when the value is not empty?

Reply:
This will not break the existing encrypted, as tested before https://github.com/longhorn/longhorn/issues/3353#issuecomment-1332344405
`luksFormat` will only initialized device once when diskformat (from mount-utils) is empty 

>Should we set these options when the value is not empty?

empty?  we will use default values for most distros we tested if users do not provide these variables.